### ISSUE TITLE
how to use this bot

### ISSUE THREAD

**Issue by sanjayradadiya** [TeamFreeHugs/Whatsapp-Chatbot#1]
@eyeballcode  how to use,
i have no idea about your scenario
can we have one example please

**Comment by sanjayradadiya** [comment]
@eyeballcode  how to config this chatbot step by step please give me solution. .

**Comment by eyeballcode** [comment]
Have you run this code yet?

```
git clone https://github.com/TeamFreeHugs/Whatsapp-Chatbot.git
cd Whatsapp-Chatbot
npm install -d
node bin/www
```

It's part of the readme

**Comment by sanjayradadiya** [comment]
ok but i have one little confusion about this code like
if some other user send me message then how i active that user message windows using clicks

**Comment by eyeballcode** [comment]
As of now you can't. It's something I want to do but the `MutationObserver` is detecting too much. You could try doing your own by observing the user list for a new message bubble.

**Comment by sanjayradadiya** [comment]
can you help me to develop this feature well

**Comment by andrecordeiro** [comment]
hello , I am trying to create the message bubbles for each user, but it is complicated.

**Comment by sanjayradadiya** [comment]
hello i have create one little code for message bubbles for each user but when i trying to switch between current user service is not responded an user message and also when i trying message again then give reply so i am stop their

**Comment by andrecordeiro** [comment]
we are in the same situation then friend. I think qu go to the telegram

**Comment by sanjayradadiya** [comment]
if this problem solved then we can start using for all users but one thing is complicated

**Comment by andrecordeiro** [comment]
@SR91 
He is friend . I 'm trying to help , but it's complicated. you could evolve?

**Comment by andrecordeiro** [comment]
@eyeballcode how do I use mysql require. an error occurs when I try to add.

**Comment by sanjayradadiya** [comment]
i am little bit nearer to solution but some problem is arise for sending message to specific user.

**Comment by andrecordeiro** [comment]
@eyeballcode how or where I can put the require ( "mysql")? I need to write the message in mysql after the sendmessage.

**Comment by andrecordeiro** [comment]
@SR91 @eyeballcode 
how should I send pictures ? I tried this way and there is a CANVAS error ... Can you help me ?

whatsapp.sendImage("https://image.freepik.com/icones-gratis/robo-de-brinquedo_318-69289.jpg");

**Comment by sanjayradadiya** [comment]
hi friends 
i have also faced this problem but solution is not found yet
this occurred due to whatsapp cross domain restrictions

**Comment by andrecordeiro** [comment]
@SR91 
I used https to host my image, and there was no more error, but does not send the image only appears in the url inspector .

**Comment by sanjayradadiya** [comment]
but if we want to use http then it may be usefull ? bcz i hanv't secure connection so what i do?

**Comment by andrecordeiro** [comment]
I just got via https , but this is due to the web browser requirement . I'm using firefox . but I can not send a picture . I think I'm doing something wrong.

**Comment by sanjayradadiya** [comment]
first of all we want to check that actually this feature is well run or not

**Comment by andrecordeiro** [comment]
hello @SR91  you could perform sending images ?

**Comment by sanjayradadiya** [comment]
no friend never yet but in feature is required

**Comment by andrecordeiro** [comment]
I tried to send an image this way: whatsapp.sendImage ( " https://www.mydomain.com/teste.jpg " ) and returned me a CORS error : Image from origin ' https://www.mydomain.com ' has Been blocked from loading by Cross-Origin Resource Sharing policy : In the ' Access -Control- Allow-Origin ' header is present on the requested resource . Origin ' https://web.whatsapp.com ' is Therefore not allowed access. @eyeballcode  i can help me?

**Comment by sanjayradadiya** [comment]
yes i have also got this type of same error while send image using this procedure.

**Comment by sanjayradadiya** [comment]
@eyeballcode 
any updates regarding switch user
we are waiting for your positive reply.

**Comment by eyeballcode** [comment]
Whoops. This is a CORS issue.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS

It's also a CSP issue.

https://developer.mozilla.org/en-US/docs/Web/Security/CSP

There's nothing I can do about it, except that you could try the requestSomething method for cross origin stuff.

**Comment by andrecordeiro** [comment]
@@@Thank you friend. You could show an example of how to sendImage using requestSomething ?

**Comment by ComedianCockleDoodleDoo** [comment]
Your persistence is inspiring. And thank you for this discovery.

sanjayradadiya commented on Apr 7, 2016
hi friends
i have also faced this problem but solution is not found yet
this occurred due to whatsapp cross domain restrictions

**Comment by eyeballcode** [comment]
@ComedianCockleDoodleDoo @andrecordeiro @sanjayradadiya 

Hello, I have re-written the API to work with Electron so the user does not have to bother with injection.

Also, this exposes all NodeJS APIs, allowing things like MongoDB to be access from the browser.

Please take a look [here](https://github.com/TeamFreeHugs/WhatsApp-Chatbot-2.0)
